File: backend/src/main/java/com/backend/agrilink/dto/LoginResponse.java
package com.backend.agrilink.dto;

import com.backend.agrilink.model.ProvinceList;
import com.backend.agrilink.model.TipoUsuario;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class LoginResponse {

    private String token;

    private String id;
    private String nome;
    private String telefone;
    private TipoUsuario tipo;
    private ProvinceList provincia;
}

File: frontend/lib/models/login_response.dart
class LoginResponse {
  final String token;
  final String id;
  final String nome;
  final String telefone;

  LoginResponse({required this.token, required this.id, required this.nome, required this.telefone});

  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    return LoginResponse(
      token: json['token'],
      id: json['id'],
      nome: json['nome'],
      telefone: json['telefone'],
    );
  }
}
